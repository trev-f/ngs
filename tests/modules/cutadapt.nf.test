nextflow_process {

    name "Test Process CUTADAPT_ADAPTERS"
    script "modules/cutadapt.nf"
    process "CUTADAPT"

    test("CUTADAPT works for single-end reads.") {

        when {
            params {
            }
            process {
                """
                input[0] = [
                    [
                        sampleName: "SRR1066657_GSM1299413_WT_NR_A",
                        readType:   "single"
                    ],
                    [
                        file("https://github.com/utia-gc/ngs-test/raw/rnaseq/data/reads/SRR1066657_GSM1299413_WT_NR_A_Saccharomyces_cerevisiae_RNA-Seq_50000.fastq.gz")
                    ]
                ]
                input[1] = "AGATCGGAAGAGCGTCGTGTAGGGAAAGAGTGT"
                input[2] = "AGATCGGAAGAGCACACGTCTGAACTCCAGTCAC"
                input[3] = "30"
                """
            }
        }

        then {
            assert process.success

            with(process.out.reads) {
                assert size() == 1
                assert get(0).get(1) ==~ /^.*\/SRR1066657_GSM1299413_WT_NR_A_R1.fastq.gz/
            }
            with(process.out.log) {
                assert size() == 1
                assert get(0) ==~ /^.*\/SRR1066657_GSM1299413_WT_NR_A_cutadapt-log.txt/
            }

            assert snapshot(process.out.reads).match()
        }

    }

}
