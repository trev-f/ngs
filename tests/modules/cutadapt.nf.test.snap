{
    "CUTADAPT works for single-end reads.": {
        "content": [
            [
                [
                    {
                        "sampleName": "SRR1066657_GSM1299413_WT_NR_A",
                        "readType": "single"
                    },
                    "SRR1066657_GSM1299413_WT_NR_A_R1.fastq.gz:md5,9c46e73133a169e413c6f94eb6afb277"
                ]
            ]
        ],
        "timestamp": "2023-07-31T19:06:39+0000"
    }
}